### PR TITLE
fix: gnome-autoar - remove flagged files

### PR DESCRIPTION
**Pull request by anphel31**
Remove three encrypted extract-test fixtures from gnome-autoar and drop the three meson test cases that read them.

gnome-autoar ships three encrypted extract-test fixtures (`tests/files/extract/test-encrypted*/input/arextract.zip`, password-protected zips). The package-signing scan can't inspect encrypted archives and rejects the .src.rpm, blocking signing. The files/cases are test-only and not shipped in any binary RPM.

`%check` runs `%meson_test`, so removing the fixtures also drops the three meson cases that read them (`test_encrypted`, `test_encrypted_request_passphrase`, `test_encrypted_wrong_passphrase` and their registrations).

Why not an azldev archive overlay (as used for the other packages)? The tarball also ships an absolute-target symlink fixture (`tests/files/extract/test-symlink-parent/reference/arextract -> /tmp`) that azldev's overlay extractor rejects while extracting the whole archive — before overlays apply. Instead, `modify_source.sh` repacks the tarball out-of-band and it's served via `origin=download`.

`azldev comp render --check-only` reports no drift.

### CODE REVIEW

File: base/comps/gnome-autoar/modify_source.sh [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+# staging lookaside.
+#
+# Output lands under <repo-root>/base/build/work/scratch/gnome-autoar/.
+
+set -euo pipefail
+
+VERSION="0.4.5"
+ORIGINAL_NAME="gnome-autoar-${VERSION}.tar.xz"
+
+# Upstream Source0 (download.gnome.org) SHA512, from specs/g/gnome-autoar/sources.
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Fixed. Clarified the comment so it's explicit that `UPSTREAM_SHA512` is the pristine upstream download.gnome.org checksum used to verify the download before repacking — and explicitly **not** the modified/served hash that now lives in `specs/g/gnome-autoar/sources`. That was the only reference to `specs/.../sources`; I didn't find a second occurrence at line 129 (the tail of the script only computes and prints the modified hash + the upload command, which is correct).

File: base/comps/gnome-autoar/gnome-autoar.comp.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# used here: the upstream tarball ships an absolute-target symlink fixture
+# (tests/files/extract/test-symlink-parent/reference/arextract -> /tmp) that
+# azldev's overlay extractor rejects, so the repack is done out-of-band and
+# served from the staging lookaside via origin=download.
+[[components.gnome-autoar.source-files]]
+filename = "gnome-autoar-0.4.5.tar.xz"
+hash = "855c016959c216b233c5a7c07c8a96f8beeb74b55154fdc054f67768c4b4935b7dc35d16dcd8aa8feee9b5ec283acaf92dd0ea686ebcd9a2a00cc6a2753da4a1"
+hash-type = "SHA512"
+origin = { type = "download", uri = "https://azltempstaginglookaside.blob.core.windows.net/repo/pkgs_modified/gnome-autoar/gnome-autoar-0.4.5.tar.xz/sha512/855c016959c216b233c5a7c07c8a96f8beeb74b55154fdc054f67768c4b4935b7dc35d16dcd8aa8feee9b5ec283acaf92dd0ea686ebcd9a2a00cc6a2753da4a1/gnome-autoar-0.4.5.tar.xz" }
+replace-upstream = true
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/gnome-autoar.comp.toml:14]: Verified via the `modify_source.sh` post-conditions and `azldev comp render --check-only` (no drift): the script checks the upstream SHA512 before editing, removes the encrypted fixtures, drops the three `test_encrypted*` cases and their `g_test_add_func` registrations, and asserts siblings (e.g. `test_readonly_directory`) survive. meson uses only selective `-Werror=` flags (no `-Werror=unused-function` and no blanket `werror`), so removing the cases doesn't break compilation. A full RPM/mock build isn't runnable in my environment; CI performs it.

File: base/comps/gnome-autoar/modify_source.sh [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+TEST_FILE="tests/test-extract-unit.c"
+
+SCRIPT_DIR="$(cd "$(dirname "$(realpath "$0")")" && pwd)"
+REPO_ROOT="$(cd "${SCRIPT_DIR}/../../.." && pwd)"
+WORKDIR="${REPO_ROOT}/base/build/work/scratch/gnome-autoar"
+mkdir -p "${WORKDIR}"
+cd "${WORKDIR}"
+
+echo "[1/6] Downloading ${ORIGINAL_NAME}"
+[[ -f "${ORIGINAL_NAME}" ]] || curl -fsSL --retry 3 -o "${ORIGINAL_NAME}" "${UPSTREAM_URL}"
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Fixed. The download now writes to `${ORIGINAL_NAME}.part` and is renamed to the final name only on success, so an interrupted transfer can't leave a truncated file that the `-f` guard would silently reuse.

File: base/comps/gnome-autoar/modify_source.sh [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+(
+    cd "${EXTRACT_DIR}"
+    tar --sort=name \
+        --mtime='2024-01-01 00:00:00 UTC' \
+        --owner=0 --group=0 --numeric-owner \
+        -cf - "${TOPDIR}"
+) | xz -T1 -9e > "${ORIGINAL_NAME}.modified"
+
+MODIFIED_SHA512=$(sha512sum "${ORIGINAL_NAME}.modified" | awk '{print $1}')
+echo "${MODIFIED_SHA512}  ${ORIGINAL_NAME}" > "${ORIGINAL_NAME}.sha512"
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Fixed. The sidecar now records the modified file's name (`${ORIGINAL_NAME}.modified`) written to `${ORIGINAL_NAME}.modified.sha512`, so `sha512sum -c` verifies against the file that was actually hashed.

File: base/comps/gnome-autoar/modify_source.sh [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    tests/files/extract/test-encrypted-wrong-passphrase
+)
+
+# The meson extract-unit test hard-codes three encrypted cases that read the
+# fixtures above; drop their function definitions and g_test_add_func
+# registrations so the remaining suite still runs.
+TEST_FILE="tests/test-extract-unit.c"
+
+SCRIPT_DIR="$(cd "$(dirname "$(realpath "$0")")" && pwd)"
+REPO_ROOT="$(cd "${SCRIPT_DIR}/../../.." && pwd)"
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: suggestion(non-blocking): `git rev-parse --show-toplevel` would probably be more reliable here.

File: base/comps/gnome-autoar/modify_source.sh [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+WORKDIR="${REPO_ROOT}/base/build/work/scratch/gnome-autoar"
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: nit/suggestion(non-blocking): I'd probably pull the various instances of `gnome-autoar` into a variable like `PACKAGE_NAME`. Definitely non-blocking; take it or leave it.

File: base/comps/gnome-autoar/modify_source.sh [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+mkdir -p "${WORKDIR}"
+cd "${WORKDIR}"
+
+echo "[1/6] Downloading ${ORIGINAL_NAME}"
+if [[ ! -f "${ORIGINAL_NAME}" ]]; then
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: issue(non-blocking): This seems overly complicated. Is the entire point of this to prevent issues when calling this script simultaneously from two different shells? That doesn't seem worth guarding against.

Consider just bailing if the file exists or even moving on to the next step -- it will check the sha256 anyway, so you'll be guaranteed to have the right file.

File: base/comps/gnome-autoar/modify_source.sh [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    if [[ ! -d "${target}" ]]; then
+        echo "ERROR: expected fixture dir not present in upstream tarball: ${rel}" >&2
+        exit 1
+    fi
+    rm -rf "${target}"
+done
+
+echo "[5/6] Dropping encrypted test cases from ${TEST_FILE}"
+export TEST_PATH="${EXTRACT_DIR}/${TOPDIR}/${TEST_FILE}"
+python3 <<'PY'
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: issue(blocking): This is a lot of (embedded) python code to get rid of three test cases. Is there a reason we wouldn't use `%meson_test --exclude test1 --exclude test2 --exclude test3`? That's a simple overlay and gets rid of all of this, unless I'm missing something.

Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: issue(blocking): If we decide to keep this approach rather than `%meson_test --exclude` (see [above comment](https://github.com/microsoft/azurelinux/pull/18626/changes#r3872939292)), it makes more sense to me to have this entire script be python rather than over half of it be embedded python.

File: base/comps/gnome-autoar/modify_source.sh [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+MODIFIED_SHA512=$(sha512sum "${ORIGINAL_NAME}.modified" | awk '{print $1}')
+# Record the checksum against the actual modified file so `sha512sum -c` works.
+echo "${MODIFIED_SHA512}  ${ORIGINAL_NAME}.modified" > "${ORIGINAL_NAME}.modified.sha512"
+
+cat <<EOF
+
+modified tarball: ${WORKDIR}/${ORIGINAL_NAME}.modified
+SHA512:           ${MODIFIED_SHA512}
+
+Upload (after \`az login\`):
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.sh]: question(blocking-on-answer): Does this actually work? I thought most people didn't have this entitlement and you could only use our pipelines to upload these files.

**Review by tobiasb-ms — CHANGES_REQUESTED** [review]

File: base/comps/gnome-autoar/modify_source.sh [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    echo "  expected: ${UPSTREAM_SHA512}" >&2
+    echo "  computed: ${computed}" >&2
+    exit 1
+fi
+
+echo "[3/6] Extracting"
+EXTRACT_DIR="extracted"
+rm -rf "${EXTRACT_DIR}"
+mkdir "${EXTRACT_DIR}"
+tar -xf "${ORIGINAL_NAME}" -C "${EXTRACT_DIR}"
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Good catch. I pinned `umask 022` before extraction so the recorded file modes no longer depend on the caller's environment. One caveat: the whole script is now Python and `xz` output still isn't guaranteed identical across liblzma versions, so the docstring no longer claims cross-toolchain byte-reproducibility — the artifact published to the lookaside (pinned by SHA-512 in `specs/g/gnome-autoar/sources`) is the source of truth the build downloads.

File: base/comps/gnome-autoar/modify_source.sh (hunk 8 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Done — the whole script is now Python (`modify_source.py`); the shell wrapper is gone and the C-file edit is native Python instead of an embedded heredoc.

File: base/comps/gnome-autoar/modify_source.sh (hunk 5 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Done — now uses `git rev-parse --show-toplevel`.

File: base/comps/gnome-autoar/modify_source.sh (hunk 6 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Done — added a `PACKAGE_NAME` constant that the paths, upstream URL and blob name all derive from.

File: base/comps/gnome-autoar/modify_source.sh (hunk 7 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Simplified — dropped the `.part`/rename dance. It just bails if the file already exists, and the SHA-512 check rejects any partial or corrupt download on a re-run.

File: base/comps/gnome-autoar/modify_source.sh (hunk 9 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: You're right — most contributors don't have write access to `azltempstaginglookaside`; that `az storage blob upload` line is the maintainer/pipeline step, not something every contributor can run. I've reworded the script's printed guidance to say so (hand the file to someone with access, or use the lookaside-upload pipeline). The artifact for this PR is already uploaded and pinned by SHA-512 in `specs/g/gnome-autoar/sources`.

File: base/comps/gnome-autoar/modify_source.sh (hunk 8 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: The `%meson_test --exclude` route doesn't solve this one: the package-signing scan blocks on the encrypted `input/arextract.zip` files physically present in the `.src.rpm`, not on test *execution*. Excluding the tests would still ship the flagged fixtures in the source, so they have to be physically removed from the tarball (hence the repack). Dropping the three test cases is only so `%check` still builds once the fixtures are gone.

Comment (tobiasb-ms) [reply base/comps/gnome-autoar/modify_source.sh]: I think I was unclear in my comment. I see that we need to remove `input/arextract.zip`. But I'm wondering if we can get away with just excluding the tests rather than deleting the tests themselves.

If we only dropped `arextract.zip` and not the actual tests, would `%meson-test --exclude ...`  fail because of a build-time dependency? Or would the actual build fail because it's referencing collateral that isn't there?

File: base/comps/gnome-autoar/modify_source.py [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+            f"  expected: {UPSTREAM_SHA512}\n"
+            f"  computed: {computed}"
+        )
+
+    print("[3/6] Extracting")
+    extract_dir = Path("extracted")
+    if extract_dir.exists():
+        shutil.rmtree(extract_dir)
+    extract_dir.mkdir()
+    run(["tar", "-xf", ORIGINAL_NAME, "-C", str(extract_dir)])
```
Comment (tobiasb-ms) [inline base/comps/gnome-autoar/modify_source.py]: nit(non-blocking): Did you consider using the `tarfile` standard library? It supports this extraction and of the repackaging (though I'm not certain about guaranteed reproducibility).

**Review by tobiasb-ms — CHANGES_REQUESTED** [review]
There are a couple other outstanding questions that need to be resolved, but the code currently looks good.

File: base/comps/gnome-autoar/modify_source.sh (hunk 8 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Good question — I checked the upstream test layout to be sure. Short answer: `%meson_test --exclude` can't target these three cases, and the failure would be at **runtime**, not build time.

- `tests/meson.build` registers a single meson `test()` target for these — `test-extract-unit`. The three encrypted cases (`test_encrypted`, `test_encrypted_request_passphrase`, `test_encrypted_wrong_passphrase`) are **GLib sub-tests** added with `g_test_add_func()` *inside* that one binary, not separate meson tests. So `--exclude` can only drop the whole `test-extract-unit` binary, not the three sub-tests.
- The fixtures are read at **runtime** (`G_TEST_SRCDIR` + `g_file_get_child (input, "arextract.zip")`), so nothing references them at compile/build time. If we dropped only `input/arextract.zip` and kept the sub-tests, the build would still succeed, but at `%check` those three sub-tests would fail on the missing file → `test-extract-unit` exits non-zero → `%meson_test` fails.
- Excluding the whole `test-extract-unit` binary would also drop the ~dozen other extract sub-tests in it that use the *non-encrypted* `arextract.zip` fixtures we keep — a real coverage loss.

So removing just the three sub-test functions + their three `g_test_add_func` registrations is the minimal change that keeps the rest of `test-extract-unit` running once the encrypted fixtures are physically removed for the signing scan. To answer your two questions directly: it wouldn't be a build-time dependency failure — it'd be `%check` failing at runtime on the missing collateral.

File: base/comps/gnome-autoar/modify_source.sh (hunk 9 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.sh]: Good catch on the help text leaking internal-process details — fixed in the latest push (0bdcc9a). The script's final message no longer prints the storage-account name, container, blob path, or the `az storage blob upload` command; it now just reports the regenerated tarball path + SHA-512 and points maintainers at the internal docs for the publishing step. The `origin=download` URI in the comp.toml is unchanged, since that's the same source-mirror pattern the other repacked components already use — happy to revisit that repo-wide separately if we want to move it behind a neutral host.

The same push also handles the Python type/lint notes (dropped the untyped `run()` wrapper in favor of direct `subprocess.run(..., check=True)`, added the missing docstrings, and narrowed the `Popen.stdout` deref) so it's ruff/pyright-clean, and hardened the download so an interrupted or corrupt fetch can't wedge later re-runs. `azldev component render --check-only gnome-autoar` reports no drift.

File: base/comps/gnome-autoar/modify_source.sh (hunk 8 above)
Comment (tobiasb-ms) [reply base/comps/gnome-autoar/modify_source.sh]: Sounds good; thanks for looking into it.

**Review by tobiasb-ms — CHANGES_REQUESTED** [review]
Changes look good, but please squash the three commits into one, with a conventional commit message (the first commit has a good message). We do rebase merges, so all of the commits in a PR end up in the history, and all of these are one logical change.

While you're at it, look at the last commit, which seems to be an empty merge from 4.0. You should be rebasing rather than merging, but since it's empty it seems kind of funky and per above paragraph you should be squashing it out anyway.

File: base/comps/gnome-autoar/modify_source.py [hunk 12]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py:1]: I can't reproduce this one. With the repo `ruff.toml` (`select = ["ALL"]`), `ruff check --select INP001 base/comps/gnome-autoar/modify_source.py` reports `All checks passed`. `INP001` fires for a file that's part of an *implicit namespace package* (sibling modules in the dir, or the file being imported as a package); this script is the only module in its directory and is executed directly (`python3 .../modify_source.py`), never imported — the same situation as the `scripts/ci` helpers that the per-file-ignore already covers — so the rule doesn't trigger.

Adding a file-level `# noqa: INP001` would actually be counter-productive: since the rule doesn't fire and `select = ALL` also enables `RUF100`, the unused suppression itself fails `ruff check` (`RUF100 Unused noqa directive`). So I've left the file as-is. The only diagnostic ruff emits on it is `CPY001`, which is a preview rule not enabled in CI (existing committed scripts trip it too), and Pyright is clean.

File: base/comps/gnome-autoar/modify_source.py (hunk 11 above)
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py]: Good thought — I went with GNU `tar` deliberately for byte-reproducibility. The repack pins `--sort=name --mtime=... --owner=0 --group=0 --numeric-owner` so member order, timestamps and ownership are fixed and the output SHA-512 is stable. Stdlib `tarfile` doesn't give an equivalent one-call guarantee: you'd have to sort the members and normalize each `TarInfo` (uid/gid/uname/gname/mtime) by hand, and even then the byte layout wouldn't necessarily match the `tar` output the served artifact was originally produced with. Since the whole point here is a stable hash, I kept GNU tar. Non-blocking, and happy to revisit if we ever move the canonical artifact generation.

File: base/comps/gnome-autoar/modify_source.py [hunk 13]
```diff
@@ -0,0 +1,227 @@
...
+    workdir = repo_root / "base" / "build" / "work" / "scratch" / PACKAGE_NAME
+    workdir.mkdir(parents=True, exist_ok=True)
+    os.chdir(workdir)
+
+    original = Path(ORIGINAL_NAME)
+    print(f"[1/6] Downloading {ORIGINAL_NAME}")
+    if not original.exists():
+        # No atomic-rename dance: the SHA-512 check below rejects any partial or
+        # corrupt download, so a re-run simply re-fetches it.
+        run(["curl", "-fsSL", "--retry", "3", "-o", ORIGINAL_NAME, UPSTREAM_URL])
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py]: Outdated — addressed in the current revision (squashed into `c90f5e5e8c`). The download now cleans up a partial file on a failed `curl` (`try/except CalledProcessError` → `original.unlink(missing_ok=True)`) and also unlinks on a checksum mismatch, so a re-run re-fetches cleanly. I used cleanup-on-failure rather than the `.part`/atomic-rename dance per an earlier request from @tobiasb-ms to keep the download path simple.

File: base/comps/gnome-autoar/modify_source.py [hunk 14]
```diff
@@ -0,0 +1,227 @@
...
+TEST_FILE = "tests/test-extract-unit.c"
+
+# Fixed umask so `tar -x` records reproducible file modes regardless of the
+# caller's environment (extracting the same archive under different umasks would
+# otherwise yield different modes and therefore a different repacked SHA-512).
+REPACK_UMASK = 0o022
+
+
+def run(cmd: list[str], **kwargs) -> subprocess.CompletedProcess:
+    return subprocess.run(cmd, check=True, **kwargs)
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py]: Outdated — fixed in `c90f5e5e8c`. The thin `run()` wrapper was removed entirely; call sites now use `subprocess.run(..., check=True)` directly, which drops the untyped `**kwargs`, the bare `CompletedProcess`, and the missing-docstring issue in one go (annotating `**kwargs: Any` wouldn't work anyway, since `select = ALL` also enables ANN401). ruff and pyright are both clean on the file.

File: base/comps/gnome-autoar/modify_source.py [hunk 15]
```diff
@@ -0,0 +1,227 @@
...
+
+
+def sha512_of(path: Path) -> str:
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py:76]: Outdated — fixed in `c90f5e5e8c`. The public helpers now have Google-style docstrings (`sha512_of`, `main`; `edit_test_file` already had one), and the `run` wrapper that lacked one was removed. `ruff check` reports no D103 on the file.

File: base/comps/gnome-autoar/modify_source.py [hunk 16]
```diff
@@ -0,0 +1,227 @@
...
+                "-",
+                TOPDIR,
+            ],
+            cwd=extract_dir,
+            stdout=subprocess.PIPE,
+        )
+        xz = subprocess.Popen(
+            ["xz", "-T1", "-9e"], stdin=tar.stdout, stdout=out
+        )
+        tar.stdout.close()  # allow tar to receive SIGPIPE if xz exits
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py:199]: Outdated — fixed in `c90f5e5e8c`. `tar.stdout` is now narrowed with an explicit `if tar.stdout is None: sys.exit(...)` guard before it's passed to `xz` and closed, so the standard-mode pyright check passes (0 errors).

File: base/comps/gnome-autoar/modify_source.py [hunk 17]
```diff
@@ -0,0 +1,224 @@
...
+    os.umask(REPACK_UMASK)
+
+    repo_root = Path(
+        subprocess.run(
+            ["git", "rev-parse", "--show-toplevel"],
+            check=True,
+            capture_output=True,
+            text=True,
+        ).stdout.strip()
+    )
```
[the remark this answers is not in the document]
Comment (anphel31) [reply base/comps/gnome-autoar/modify_source.py]: Good catch — fixed in `c36a5404c6`. Switched repo-root resolution from `git rev-parse --show-toplevel` (which keys off the caller's CWD) to `Path(__file__).resolve().parents[3]`, so it resolves this script's own checkout even when invoked by absolute path from another directory. This also matches the convention the other repack scripts use (firefox/gdal/espeak-ng/exfatprogs derive `REPO_ROOT` from `BASH_SOURCE`).

Heads up @tobiasb-ms: this reverses your earlier non-blocking suggestion to use `git rev-parse` — I went with `__file__` here for the CWD-robustness + consistency with the sibling scripts, but happy to switch back if you'd prefer.

[the remark this answers is not in the document]
Comment (tobiasb-ms) [reply base/comps/gnome-autoar/modify_source.py]: Thanks for the heads up. I don't think we need to bend over backwards to use `git rev-parse` here.

**Review by tobiasb-ms — APPROVED** [review]
